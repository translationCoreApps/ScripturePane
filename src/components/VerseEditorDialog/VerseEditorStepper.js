import React from 'react';
import PropTypes from 'prop-types';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { Step, Stepper, StepLabel } from "material-ui/Stepper";

/**
 * Renders the steps for editing the verse
 */
class VerseEditorStepper extends React.Component {
  render() {
    const {stepIndex, steps} = this.props;

    return (
      <MuiThemeProvider>
        <Stepper activeStep={stepIndex}>
          {steps.map((step, index) => {
            return (
              <Step key={index}>
                <StepLabel>{step}</StepLabel>
              </Step>
            );
          })}
        </Stepper>
      </MuiThemeProvider>
    );
  }
}

VerseEditorStepper.propTypes = {
  stepIndex: PropTypes.number,
  steps: PropTypes.string
};

export default VerseEditorStepper;
