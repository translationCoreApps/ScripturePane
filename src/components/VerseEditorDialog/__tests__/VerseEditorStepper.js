/* eslint-env jest */

import React from 'react';
import VerseEditorStepper from '../VerseEditorStepper';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import {render} from 'enzyme';

describe('<VerseEditorStepper/>', () => {


  it('on first step', () => {
    const wrapper = render(
      <MuiThemeProvider>
        <VerseEditorStepper stepIndex={0} steps={['step 1', 'step 2']}/>
      </MuiThemeProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('on second step', () => {
    const wrapper = render(
      <MuiThemeProvider>
        <VerseEditorStepper stepIndex={1} steps={['step 1', 'step 2']}/>
      </MuiThemeProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });

});
