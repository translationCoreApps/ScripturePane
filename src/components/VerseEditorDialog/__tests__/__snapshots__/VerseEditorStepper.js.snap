// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VerseEditorStepper/> on first step 1`] = `
<div
  style="display:flex;flex-direction:row;align-content:center;align-items:center;justify-content:space-between;mui-prepared:"
>
  <div
    style="flex:0 0 auto;mui-prepared:"
  >
    <span
      style="height:72px;color:rgba(0, 0, 0, 0.87);display:flex;align-items:center;font-family:Roboto, sans-serif;font-size:14px;padding-left:14px;padding-right:14px;font-weight:500;mui-prepared:"
    >
      <span
        style="padding-right:8px;mui-prepared:"
      >
        <svg
          style="display:block;color:#00bcd4;fill:#00bcd4;height:24px;width:24px;user-select:none;transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;font-size:24px;mui-prepared:"
          viewBox="0 0 24 24"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <text
            fill="#fff"
            font-size="12"
            text-anchor="middle"
            x="12"
            y="16"
          >
            1
          </text>
        </svg>
      </span>
      step 1
    </span>
  </div>
  <div
    style="flex:1 1 auto;mui-prepared:"
  >
    <span
      style="display:block;border-color:#bdbdbd;margin-left:-6px;border-top-style:solid;border-top-width:1px;mui-prepared:"
    />
  </div>
  <div
    style="flex:0 0 auto;margin-left:-6px;mui-prepared:"
  >
    <span
      disabled=""
      style="height:72px;color:rgba(0, 0, 0, 0.26);display:flex;align-items:center;font-family:Roboto, sans-serif;font-size:14px;padding-left:14px;padding-right:14px;cursor:default;mui-prepared:"
    >
      <span
        style="padding-right:8px;mui-prepared:"
      >
        <svg
          style="display:block;color:#9e9e9e;fill:#9e9e9e;height:24px;width:24px;user-select:none;transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;font-size:24px;mui-prepared:"
          viewBox="0 0 24 24"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <text
            fill="#fff"
            font-size="12"
            text-anchor="middle"
            x="12"
            y="16"
          >
            2
          </text>
        </svg>
      </span>
      step 2
    </span>
  </div>
</div>
`;

exports[`<VerseEditorStepper/> on second step 1`] = `
<div
  style="display:flex;flex-direction:row;align-content:center;align-items:center;justify-content:space-between;mui-prepared:"
>
  <div
    style="flex:0 0 auto;mui-prepared:"
  >
    <span
      style="height:72px;color:rgba(0, 0, 0, 0.87);display:flex;align-items:center;font-family:Roboto, sans-serif;font-size:14px;padding-left:14px;padding-right:14px;mui-prepared:"
    >
      <span
        style="padding-right:8px;mui-prepared:"
      >
        <svg
          style="display:block;color:#00bcd4;fill:#00bcd4;height:24px;width:24px;user-select:none;transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;font-size:24px;mui-prepared:"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
      </span>
      step 1
    </span>
  </div>
  <div
    style="flex:1 1 auto;mui-prepared:"
  >
    <span
      style="display:block;border-color:#bdbdbd;margin-left:-6px;border-top-style:solid;border-top-width:1px;mui-prepared:"
    />
  </div>
  <div
    style="flex:0 0 auto;margin-left:-6px;mui-prepared:"
  >
    <span
      style="height:72px;color:rgba(0, 0, 0, 0.87);display:flex;align-items:center;font-family:Roboto, sans-serif;font-size:14px;padding-left:14px;padding-right:14px;font-weight:500;mui-prepared:"
    >
      <span
        style="padding-right:8px;mui-prepared:"
      >
        <svg
          style="display:block;color:#00bcd4;fill:#00bcd4;height:24px;width:24px;user-select:none;transition:all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;font-size:24px;mui-prepared:"
          viewBox="0 0 24 24"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <text
            fill="#fff"
            font-size="12"
            text-anchor="middle"
            x="12"
            y="16"
          >
            2
          </text>
        </svg>
      </span>
      step 2
    </span>
  </div>
</div>
`;
